src/assets/JourneyWithGo-ABlogSeries/goFundamentals.md

```diff
@@ -0,0 +1,972 @@
+
+### **6.1. Method Definition**
+
+A method can be defined using the following syntax:
+
+**Syntax:**
+
+```go
+func (<receiver>) <method_name>(<comma-separated-parameters>) <return_type> {
+  // method body is the code that is executed when the method is called
+}
+```
+
+**Examples:**
+
+```go
+package main
+
+type Triangle struct {
+	  base   float64
+	  height float64
+}
+
+// defining a method named `area` that takes no parameters and returns a float64
+func (t Triangle) area() float64 {
+  return 0.5 * t.base * t.height
+}
+```
+
+In the above example we have defined a method associated with a struct named `Triangle` by adding a receiver to the function definition. The receiver is the name of the struct that the method is associated with. The receiver is enclosed in parentheses before the method name.
+
+### **6.2. Method Call**
+
+A method can be called using the following syntax:
+
+**Syntax:**
+
+```go
+<receiver>.<method_name>(<comma-separated-arguments>)
+```
+
+**Examples:**
+
+```go
+package main
+
+import (
+  "fmt"
+)
+
+type Triangle struct {
+	  base   float64
+	  height float64
+}
+
+func (t Triangle) area() float64 {
+  return 0.5 * t.base * t.height
+}
+
+func main() {
+  triangle := Triangle{base: 10, height: 5}
+  fmt.Println(triangle)
+
+  // calling the `area` method
+  area := triangle.area()                                           // triangle is the receiver
+  fmt.Println("Area of triangle is " + fmt.Sprintf("%.2f", area))
+}
+```
+
+**Output:**
+
+```bash
+{10 5}
+Area of triangle is 25.00
+```
+
+### **6.3. Function Vs Method**
+
+A function is a block of code that performs a specific task whereas a method is a function that is associated with a type. We can create multiple methods of same name associated to different types but the function name must be different (see example of [7.2](#L783)).
+
+Let's see the difference between a function and a method with the help of an example.
+
+**Examples:**
+
+```go
+package main
+
+import (
+  "fmt"
+)
+
+// defining a function named `area` that takes two float64 parameters and returns a float64
+func getAreaOfTriangle(base float64, height float64) float64 {
+  return 0.5 * base * height
+}
+
+type Triangle struct {
+    base   float64
+    height float64
+}
+
+// defining a method named `area` that takes no parameters and returns a float64
+func (t Triangle) getAreaOfTriangle() float64 {
+  return 0.5 * t.base * t.height
+}
+
+func main() {
+  // calling the `area` function
+  area := getAreaOfTriangle(10, 5)
+  fmt.Println("Area of triangle is " + fmt.Sprintf("%.2f", area))
+
+  triangle := Triangle{base: 10, height: 5}                         // creating a new instance of the `Triangle` struct
+  fmt.Println(triangle)
+
+  // calling the `area` method
+  area = triangle.getAreaOfTriangle()                               // triangle is the receiver
+  fmt.Println("Area of triangle is " + fmt.Sprintf("%.2f", area))
+}
+```
+
```

Comment:
make sure that the function names in the comments match with the actual function names